### ISSUE TITLE
The SSL connection could not be established

### ISSUE THREAD

**Issue by saed-leghaee** [dotnet-architecture/eShopOnContainers#1309]
hi
on first step of when i try to run docker-compose build command
**docker-compose build**  from step42

![Untitled](https://user-images.githubusercontent.com/64362030/80312969-7bb29c00-87fd-11ea-844c-45246e9426d3.jpg)

 many error like this was happen

 Connection refused
  Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/serilog.sinks.seq/index.json'.
  The SSL connection could not be established, see inner exception.
    Authentication failed because the remote party has closed the transport stream.

**Comment by simonmatt** [comment]
I've encountered the same issue, and retried many times, but some dependencies can be restored successfully, some cannot, how can i fix this?
![图片](https://user-images.githubusercontent.com/5940103/80335678-a1848300-8887-11ea-9149-2cf80b59ceb6.png)

**Comment by domicioam** [comment]
Are you guys behind a proxy or something? Try building the project with the error manually in Visual Studio and see if you get the same errors.

**Comment by mvelosop** [comment]
Hi @saed-leghaee, @simonmatt,

It looks like you have some firewall issue.

You should be able to explore those address in the browser like this, and should be able to verify the certificate as valid:

![image](https://user-images.githubusercontent.com/7357806/82933531-f4825080-9f81-11ea-82fe-5e7e1578d24a.png)

I guess you don't have Fiddler active in your computer, right?

Hope this helps.

**Comment by simonmatt** [comment]
@mvelosop Thanks so much for your replying to us! 
When I ran `docker-compose up` command-line in powershell directly, the issue occurred, but when I opened the solution in Visual Studio and run `Docker Compose` tasks within VS2019, the images could be pulled and containers run successfully, the MVC website can be opened as well.
Anyway, the **eshoponcontainers** app can work fine on my side except **WebSPA** app.

BTW, I've checked the certification state, it's shown as 'valid' and works fine.

-----
```bash
$ docker images
REPOSITORY                              TAG                                              IMAGE ID            CREATED             SIZE
netcore_docker_rabbitmq_worker          latest                                           c2f453c048db        4 days ago          262MB
netcore_docker_rabbitmq_publisher_api   latest                                           e7f5743eaaf5        4 days ago          261MB
mssql-linux                             latest                                           5a3b785b6bdc        6 days ago          1.35GB
rabbitmq                                3-management                                     867da7fcdf92        7 days ago          181MB
eshop/webmvc                            dev                                              84cf83274ea3        8 days ago          207MB
eshop/ordering.signalrhub               dev                                              0311c12b908c        8 days ago          207MB
eshop/webshoppingagg                    dev                                              47ab62c83d10        8 days ago          207MB
eshop/mobileshoppingagg                 dev                                              7d760b51d0ee        8 days ago          207MB
eshop/ordering.api                      dev                                              d673d2ce82a4        8 days ago          207MB
eshop/basket.api                        dev                                              06f5705716a6        8 days ago          207MB
eshop/marketing.api                     dev                                              7ae438346fdd        8 days ago          207MB
eshop/identity.api                      dev                                              9e6a9369f43d        8 days ago          207MB
eshop/catalog.api                       dev                                              db0bce97655e        8 days ago          207MB
eshop/webhooks.client                   dev                                              ef3eef81372c        8 days ago          207MB
eshop/webhooks.api                      dev                                              29469f63a414        8 days ago          207MB
eshop/ordering.backgroundtasks          dev                                              ab0918df92cf        8 days ago          207MB
eshop/webspa                            dev                                              da9c9b4190fc        9 days ago          207MB
eshop/locations.api                     dev                                              7a787850608c        9 days ago          207MB
eshop/webstatus                         dev                                              f05e28e29a71        9 days ago          207MB
eshop/payment.api                       dev                                              7dd39fdc9283        9 days ago          207MB
redis                                   latest                                           987b78fc9e38        9 days ago          104MB
dockercomposestarter_web                latest                                           46ff276872a3        12 days ago         104MB
mcr.microsoft.com/dotnet/core/aspnet    3.1                                              4bee399eb313        13 days ago         207MB
rabbitmq                                3-management-alpine                              087a668aff16        2 weeks ago         140MB
redis                                   alpine                                           360360313017        3 weeks ago         31.6MB
python                                  3.6-alpine                                       92a6cc6f3ecc        4 weeks ago         92.8MB
mongo                                   latest                                           3f3daf863757        4 weeks ago         388MB
mcr.microsoft.com/mssql/server          2017-latest                                      d04f0f18f395        6 weeks ago         1.46GB
portainer/portainer                     latest                                           2869fc110bf7        2 months ago        78.6MB
docker/desktop-kubernetes               kubernetes-v1.16.5-cni-v0.7.5-critools-v1.15.0   a86647f0b376        4 months ago        279MB
datalust/seq                            latest                                           6be4839d8ddd        4 months ago        388MB
hello-world                             latest                                           bf756fb1ae65        4 months ago        13.3kB
microsoft/dotnet                        2.2-sdk                                          2357b6790b9d        5 months ago        1.74GB
microsoft/dotnet                        2.2-aspnetcore-runtime                           e7e3b238011c        5 months ago        261MB
docker/kube-compose-installer           v0.4.25-alpha1                                   2a71ac5a1359        7 months ago        42.3MB
envoyproxy/envoy                        v1.11.1                                          d3340c53fdcd        9 months ago        151MB
```

**Comment by sughosneo** [comment]
Hi @simonmatt , as original question has been answered already, I am closing this issue as of now. Please feel free to reopen it if needed.

Thank you.

**Comment by DaveLHX** [comment]
I'm on windows 10, docker with WSL2, running linux containers.

I get the exact same behaviour any time I try to build with docker-compose, as stated above it builds in Visual studio.    
For example I build only one service in this command just so its easier to debug.
` docker-compose build --no-cache webmvc`
and always leads to this no matter what kind of cleanup I do prior.
`Step 42/50 : RUN dotnet restore "eShopOnContainers-ServicesAndWebApps.sln"
 ---> Running in ca15bf9d0d01
  Determining projects to restore...
  Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/aspnetcore-tools/nuget/v3/flatcontainer/aspnetcore.healthchecks.ui.client/index.json'.
  The SSL connection could not be established, see inner exception.
    Authentication failed because the remote party has closed the transport stream.
  Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/aspnetcore-dev/nuget/v3/flatcontainer/microsoft.extensions.configuration.azurekeyvault/index.json'.
  The SSL connection could not be established, see inner exception.
    Authentication failed because the remote party has closed the transport stream.
  Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.io.compression/index.json'.
  The SSL connection could not be established, see inner exception.`

So the previous answer does not fit the symptoms.   

![image](https://user-images.githubusercontent.com/12000766/99329756-e1737d80-284c-11eb-9ff4-b801df457dff.png)

**Comment by DaveLHX** [comment]
Ok well for anyone facing this, it seems caused by the linux image version(maybe) and Nuget
Adding --disable-parallel works but is super slow...like....really slow.
https://github.com/NuGet/Home/issues/9020
![image](https://user-images.githubusercontent.com/12000766/99341176-65326780-2857-11eb-9572-e625c63bf6a6.png)

This is what happens if I dont use --disable-parallel
![image](https://user-images.githubusercontent.com/12000766/99342370-b3e10100-2859-11eb-8609-3dd6e09c2a2c.png)

and my speed test
![image](https://user-images.githubusercontent.com/12000766/99342540-1c2fe280-285a-11eb-84ab-b29a3adf34d0.png)
 
I,m surprised the devs dont face this problem.

**Comment by CRodriguez25** [comment]
I'm having this exact same issue. disable-parallel seems to make it more consistent but wow is it slow...

**Comment by mgkeen** [comment]
Also been having this issue for ~ 2 weeks when building using the .net core docker images. disable-parallel seems to at least let it build 1/2 the time, but doesn't always work.

**Comment by maarten-kieft** [comment]
I ran in the same issue. Running the docker-compose up I run into this issue. When I run the docker builds individually there is no problem.

I think the problem is that the nuget server just receives too many requests and therefore is dropping https connections. That's why the disabled-parallel works (in some cases). When I lowered the amount of projects in my docker compose file I did throw some errors but after a few retries from dotnet restore the whole compose went up.

I'm not sure how to resolve this one though. You want to throttle / cache your traffic to the nuget server.
